N1_IntroToMonteCarlo.ipynb

```diff
@@ -0,0 +1,807 @@
+        "  # and the transition probability is greater than 0.4\n",
+        "  if current_state == and p> :\n",
+        "\n",
+        "    # stay in state 'A'\n",
+        "    current_state = \n",
+        "  # --- STATE MACHINE CODE ENDS --- #\n",
+        "\n",
+        "  # Stop updating when the system has transitioned to the end state\n",
+        "  if current_state == end_state:\n",
+        "    break\n",
+        "\n",
+        "print(\"Ending state achieved after \"+str(step+1)+\" iterations\")\n"
+      ],
+      "metadata": {
+        "id": "llr1UfKcM0j-"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Store your results here:\n",
+        "\n",
+        "data_EtoA = []#STORE YOUR VALUES IN THIS LIST\n",
+        "mean_EtoA = #<WRITE YOUR AVERAGE VALUE CALCULATION HERE>\n",
+        "stderr_EtoA = #<WRITE YOUR STANDARD ERROR CALCULATION HERE>\n",
+        "\n",
+        "print(f\"# of tries to transition from E to A: {mean_EtoA} +/- {stderr_EtoA}\")\n",
+        "\n",
+        "data_AtoE = []#STORE YOUR VALUES IN THIS LIST\n",
+        "mean_AtoE = #<WRITE YOUR AVERAGE VALUE CALCULATION HERE>\n",
+        "stderr_AtoE = #<WRITE YOUR STANDARD ERROR CALCULATION HERE>\n",
+        "\n",
+        "print(f\"# of tries to transition from A to E: {mean_AtoE} +/- {stderr_AtoE}\")"
+      ],
+      "metadata": {
+        "id": "uAwPwYbcd3yw"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "b) The program above can be written much more simply, and the experiment can be done more efficiently by defining a function for the whole experiment that lets the user automate the results. \n",
+        "\n",
+        "\n",
+        "*   Define a new function \"MarkovChainExperiment\" (outlined for you below) that takes as an inputs the number of times to run the experiment, the initial state, and the final state and returns a list containing how many steps it took to change the state for each time the experiment was run.  The function should let the user specify how many times to simulate the Markov Chain, run the simulation that number of times, and then return all of the results at once.  *Hint*: You will need to add another for loop.\n",
+        "\n",
+        "*   Rewrite the state machine code using only two if-statements\n",
+        "\n",
+        "*   Test your function by calling it in a new cell, and making a histogram of the results for 1 million experiments each of \"E to A\" and \"A to E\".\n"
+      ],
+      "metadata": {
+        "id": "gAy0LyufwT6p"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "\n",
+        "def MarkovChainExperiment():\n",
+        "\n",
+        "  # Begin by initializing the \n",
+        "  # current state with the value passed into the function\n",
+        "  current_state = \n",
```

Comment:
It might be good to also include `final_state=` here. Or better, put them as parameters to the function `MarkovChainExperiment(current_state, final_state)`?

Reply:
I see now that was the problem prompt... Sorry